src/operator/switchMapTo.ts

```diff
@@ -69,14 +67,23 @@ class SwitchMapToSubscriber<T, R, R2> extends OuterSubscriber<T, R> {
              innerSub: InnerSubscriber<T, R>): void {
     const { resultSelector, destination } = this;
     if (resultSelector) {
-      const result = tryCatch(resultSelector)(outerValue, innerValue, outerIndex, innerIndex);
-      if (result === errorObject) {
-        destination.error(errorObject.e);
-      } else {
-        destination.next(result);
-      }
+      this.trySelectResult(outerValue, innerValue, outerIndex, innerIndex);
     } else {
       destination.next(innerValue);
     }
   }
+
+  private trySelectResult(outerValue: T, innerValue: R,
```

Comment:
Small thing: can this be `tryResultSelector` since it matches up to `resultSelector`?